src/main/java/org/codehaus/groovy/ast/ASTNode.java

```diff
@@ -50,8 +50,18 @@
     private int lastColumnNumber = -1;
     private Map metaDataMap = null;
 
+    public void accept(GroovyCodeVisitor visitor) {
+        throw new RuntimeException("No accept() method implemented for class: " + getClass().getName());
+    }
+
+    /**
+     * An alias method for {@link org.codehaus.groovy.ast.ASTNode#accept(GroovyCodeVisitor)}
+     * Note: the method will be removed in a future version, e.g. Groovy 4 or Groovy 5
```

Comment:
Why not @`Deprecate` it then?

Reply:
In order to not disturb groovy users, we even do not add `@Deprecated` to the alias method `visit`.

Reply:
I was asking about `visit` method only, the one planned for removal. 
Wouldn't missing method (without prior warning) be more disturbing?

Reply:
As @paulk-asert said, "Once we have Java 9 as our minimum, we can use the `since` and `forRemoval` Deprecated attributes
and that will allow us to make such deprecations a little clearer. "

We won't remove it without any prior warnings, no worries ;-)
BTW, I believe the alias method `visit` will be reserved for a very long time.